helpers/misc_test.go

```diff
@@ -75,6 +77,24 @@ func (suite *TestMiscSuite) TestRetry() {
 	suite.Equal(3, logCount)
 }
 
+func TestContextCancellation(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	count := 0
+	fn := func() error {
+		count++
+		return errors.New("Not failing from context cancelled")
+	}
+	logger := func(_ string, _ ...any) {}
+
+	waitThatsNotCalled := 1000000 * time.Minute
+	err := RetryWithContext(ctx, fn, 5, waitThatsNotCalled, logger)
+	require.Equal(t, 0, count)
+	if !errors.Is(err, context.Canceled) {
+		t.Errorf("Expected context cancellation error, but it wasn't")
+	}
+}
```

Comment:
Suggest making this a test function `TestRetryWithContext` and have subtests within it that test the context cancellation path _and_ retries